bug: scripts/deploy-workload.sh uses unsafe xargs env loading — breaks on .env values with spaces or quotes

## Summary

\`scripts/deploy-workload.sh\` loads \`.env\` variables using \`export \$(grep | xargs)\` (line 29). This pattern is broken for any \`.env\` value that contains spaces, quotes, dollar signs, or other shell-special characters — which is common in passwords and Portainer API keys.

## Root Cause

In \`scripts/deploy-workload.sh\` lines 27–30:
```bash
if [ -f "$PROJECT_DIR/.env" ]; then
  # shellcheck disable=SC2046
  export $(grep -v '^#' "$PROJECT_DIR/.env" | xargs)
fi
```

The \`xargs\` approach:
1. **Breaks on values with spaces** — e.g. \`DASHBOARD_PASSWORD=my password\` causes xargs to split into two words, corrupting the export
2. **Breaks on quoted values** — e.g. \`JWT_SECRET="abc def"\` — the quotes are not stripped by xargs in this context; behaviour is shell-dependent
3. **Breaks on values with \`$\` or backticks** — shell expansion occurs inside xargs, potentially causing command injection or silent substitution
4. **Missing \`set -o pipefail\`** — the script has \`set -e\` but not \`pipefail\`, so if \`grep\` fails (e.g. empty .env), the error is silently swallowed and the \`export\` of an empty string still runs

The \`# shellcheck disable=SC2046\` comment acknowledges shellcheck warns about this exact pattern but suppresses it instead of fixing it.

## Steps to Reproduce

1. Add a value with a space to \`.env\`: \`PORTAINER_API_KEY=key with spaces\`
2. Run \`./scripts/deploy-workload.sh start\`
3. Observe: \`PORTAINER_API_KEY\` is set to only \`key\`, remaining words treated as separate exports → unexpected behaviour or Portainer auth failure

## Expected Behavior

All \`.env\` values are loaded correctly regardless of spaces, quotes, or special characters.

## Actual Behavior

Values with spaces or special characters are silently truncated or misinterpreted, causing unexpected failures when interacting with Portainer API (authentication fails or wrong URL used).

## Fix Approach

Replace the unsafe xargs pattern with a safe per-line loader:
```bash
if [ -f "$PROJECT_DIR/.env" ]; then
  while IFS='=' read -r key value; do
    # Skip comments and empty lines
    [[ "$key" =~ ^[[:space:]]*# ]] && continue
    [[ -z "$key" ]] && continue
    # Strip surrounding quotes from value
    value="${value%\"}"
    value="${value#\"}"
    value="${value%\'}"
    value="${value#\'}"
    export "$key=$value"
  done < "$PROJECT_DIR/.env"
fi
```

Also add \`set -o pipefail\` to the existing \`set -e\` at line 16 to catch pipeline failures.

## Relevant Files

- \`scripts/deploy-workload.sh\` — line 16 (\`set -e\`, missing \`pipefail\`), lines 27–30 (unsafe env loading)

## Acceptance Criteria

- [ ] \`.env\` values with spaces are loaded correctly
- [ ] \`.env\` values with surrounding quotes are stripped correctly
- [ ] \`set -o pipefail\` added alongside \`set -e\`
- [ ] \`shellcheck disable\` comment removed (fix is clean, not suppressed)
- [ ] \`scripts/lint-bash.sh\` passes (shellcheck or \`bash -n\`) after the fix
- [ ] Script behaviour is unchanged for well-formed \`.env\` files

## Scope
- **Severity**: high
- **Size**: small

---
*Created by autonomous analyst agent*

## Analyst Investigation

**Directories analyzed**: `scripts/deploy-workload.sh`

**Confirmed**: The unsafe xargs env loading pattern is at lines 27–30. The `# shellcheck disable=SC2046` comment explicitly acknowledges shellcheck flags this pattern but suppresses the warning.

**Additional findings from full script analysis**:

1. **Line 16**: `set -e` is present but **`set -o pipefail` is missing** — pipeline failures in grep+xargs chain are silently ignored
2. **Line 29**: The xargs word-splitting issue is well-described in the issue. Confirmed that .env values with spaces, quotes, or dollar signs will break silently
3. **Line 43**: `get_endpoint_id` returns only `.[0].Id` — if the first endpoint is null, it returns the literal string "null" which passes the empty check but fails later API calls
4. **Line 48-50**: `get_stack_id` jq filter `select(.Name == "$name")` is safe from injection since jq handles variable interpolation, but it uses double-quotes which could be confused with shell double-quotes — minor readability risk only
5. **Line 226**: `delete_one_stack` uses `curl -s -X DELETE` without checking the response — delete errors are silently ignored. A stack may fail to delete but the script prints "Deleted" regardless

**Missing error handling**:
- No check that `jq` is installed (required for all curl response parsing)
- No check that `curl` supports `-f` (fail on HTTP error) — the script uses `-s` (silent) only for most calls, not `-sf`

**Priority for the xargs fix**: High. A .env with `PORTAINER_API_KEY` containing a space will cause every stack operation to fail with confusing auth errors.

*Analyzed by autonomous analyst agent — 2026-03-16*

## AI Review — Issue #1015

### Feasibility: Ready

### Findings
- **Confirmed**: Line 29 of `scripts/deploy-workload.sh` uses `export $(grep -v '^#' "$PROJECT_DIR/.env" | xargs)` exactly as described. The `# shellcheck disable=SC2046` on line 28 suppresses the shellcheck warning instead of fixing the problem.
- **Confirmed**: Line 16 has `set -e` but is missing `set -o pipefail`. The other two shell scripts in the repo (`scripts/lint-bash.sh`, `scripts/test-pagination.sh`) both use `set -euo pipefail`, making `deploy-workload.sh` the inconsistent outlier.
- **Pattern is isolated**: No other script in the repository uses the `export $(grep | xargs)` pattern — this is the only affected file.
- **Real-world breakage is plausible**: The project's own `docker/.env.example` contains values that would break under this pattern, such as `SMTP_FROM=AI Portainer Dashboard <notifications@example.com>` (spaces and angle brackets) and `HARBOR_ROBOT_NAME=robot$dashboard` (`$` triggering shell expansion).
- **Proposed fix is correct**: The `while IFS='=' read -r key value` pattern safely handles values with spaces, quotes, and special characters. The `read` builtin assigns everything after the first `=` to the `value` variable, so values containing `=` (common in base64 secrets) are also handled correctly.
- **No conflicting issues**: Related open issues (#1017, #1024, #1026) target workload compose files and do not touch the env-loading logic in this script.

### Files Affected
- `scripts/deploy-workload.sh` — Line 16: add `pipefail` to existing `set -e`. Lines 27-30: replace unsafe xargs loader with safe `while read` loop, remove `shellcheck disable` comment.

### Risks / Concerns
- None. The fix is a direct replacement of a known-unsafe pattern with a well-established safe alternative. The script's behavior for well-formed `.env` files (no spaces, no quotes, no special chars) is unchanged.

### Dependencies
- None (no new packages or tools required).
- Not blocked by any other issue.

### Recommendation
Ready to implement. This is a small, well-defined fix isolated to a single file. The root cause is confirmed, the proposed solution is correct, and all acceptance criteria are testable. The `scripts/lint-bash.sh` shellcheck gate should pass cleanly once the `shellcheck disable` directive is removed alongside the unsafe pattern.

--- a/scripts/deploy-workload.sh
+++ b/scripts/deploy-workload.sh
@@ -27,7 +27,7 @@ EXTERNAL_NETWORKS=("app-frontend-net" "app-backend-net")
 if [ -f "$PROJECT_DIR/.env" ]; then
   while IFS='=' read -r key value; do
     # Skip comments and empty lines
-    [[ -z "$key" || "$key" =~ ^# ]] && continue
+    [[ -z "$key" || "$key" =~ ^[[:space:]]*(#|$) ]] && continue
     # Remove surrounding quotes from value
     value="${value%\"}"
     value="${value#\"}"
@@ -38,7 +38,7 @@ if [ -f "$PROJECT_DIR/.env" ]; then
 fi
 
 PORTAINER_URL="${PORTAINER_API_URL:-http://localhost:9000}"
-API_KEY="${PORTAINER_API_KEY}"
+API_KEY="${PORTAINER_API_KEY:-}"
 
 if [ -z "$API_KEY" ]; then
   echo "Error: PORTAINER_API_KEY not set in .env file"

--- a/scripts/deploy-workload.sh
+++ b/scripts/deploy-workload.sh
@@ -13,7 +13,7 @@
 #   app-frontend-net — web-platform <-> issue-simulators (net-chatter)
 #   app-backend-net  — data-services <-> workers <-> web-platform (gateway)
 
-set -e
+set -euo pipefail
 
 SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 PROJECT_DIR="$(dirname "$SCRIPT_DIR")"
@@ -23,10 +23,18 @@ COMPOSE_DIR="$PROJECT_DIR/workloads"
 STACK_NAMES=("data-services" "web-platform" "workers" "staging-dev" "issue-simulators")
 EXTERNAL_NETWORKS=("app-frontend-net" "app-backend-net")
 
-# Load environment variables
+# Load environment variables (safe: handles spaces, quotes, and special chars)
 if [ -f "$PROJECT_DIR/.env" ]; then
-  # shellcheck disable=SC2046
-  export $(grep -v '^#' "$PROJECT_DIR/.env" | xargs)
+  while IFS='=' read -r key value; do
+    # Skip comments and empty lines
+    [[ -z "$key" || "$key" =~ ^# ]] && continue
+    # Remove surrounding quotes from value
+    value="${value%\"}"
+    value="${value#\"}"
+    value="${value%\'}"
+    value="${value#\'}"
+    export "$key=$value"
+  done < "$PROJECT_DIR/.env"
 fi
 
 PORTAINER_URL="${PORTAINER_API_URL:-http://localhost:9000}"

--- a/scripts/deploy-workload.sh
+++ b/scripts/deploy-workload.sh
@@ -25,7 +25,7 @@ EXTERNAL_NETWORKS=("app-frontend-net" "app-backend-net")
 
 # Load environment variables (safe: handles spaces, quotes, and special chars)
 if [ -f "$PROJECT_DIR/.env" ]; then
-  while IFS='=' read -r key value; do
+  while IFS='=' read -r key value || [[ -n "$key" ]]; do
     # Skip comments and empty lines
     [[ -z "$key" || "$key" =~ ^[[:space:]]*(#|$) ]] && continue
     # Remove surrounding quotes from value

